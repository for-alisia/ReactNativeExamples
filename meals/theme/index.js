export default {
  colors: {
    primary: '#4a148c',
    accentColor: '#ff6f00',
    light: 'white'
  },
  randomColors: [
    '#f5428d',
    '#f54242',
    '#f5a442',
    '#f5d142',
    '#368dff',
    '#41d95d',
    '#46BFC7',
    '#439027',
    '#AD4CAD',
    '#47fced',
  ],
};
