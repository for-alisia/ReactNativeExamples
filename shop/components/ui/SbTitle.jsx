// Dependencies
import React from 'react';
import { StyleSheet, View } from 'react-native';

const SbTitle = () => {
  return <View style={styles.container}>Title Component</View>;
};

const styles = StyleSheet.create({
  container: {},
});

export default SbTitle;
