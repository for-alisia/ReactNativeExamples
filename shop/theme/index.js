const theme = {
  colors: {
    primary: '#DC8080',
    primaryLight: '#ECBDBC',
    secondary: '#D8C496',
    secondaryLight: '#F8ECC9',
    accent: '#CA5454',
    accentDark: '#6B5344',
    light: '#F6F6F6',
    white: '#ffffff',
    grey: '#685E59',
    greyLight: '#ADA19B',
    dark: '#3C3735',
    yellow: '#FF9921',
  },

  fonts: {
    montserratReg: 'montserrat-regular',
    monserratBold: 'montserrat-bold',
    playfair: 'playfair',
    playfairBold: 'playfair-bold',
  },

  fontSize: {
    xs: 12,
    s: 16,
    m: 24,
    l: 32,
    xl: 102,
  },

  padding: {
    xs: 4,
    s: 8,
    m: 12,
    l: 16,
    xl: 24,
    xxl: 32,
  },

  margin: {
    xs: 4,
    s: 8,
    m: 12,
    l: 16,
    xl: 24,
    xxl: 32,
  },

  borderRadius: {
    xs: 4,
    s: 8,
    m: 16,
    l: 24,
  },

  activeOpacity: 0.75,
};

export default theme;
