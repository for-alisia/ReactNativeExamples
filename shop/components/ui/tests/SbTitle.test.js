// @ts-nocheck
// Dependencies
import React from 'react';
import { render } from '@testing-library/react-native';
import { toHaveStyle } from '@testing-library/jest-native';

// Component
import SbTitle from '../SbTitle';

// Theme
import theme from '../../../theme';

describe('SbTitle UI Component', () => {
  // To use toHaveStyle
  expect.extend({ toHaveStyle });
  const testText = 'Hello';

  it('renders a text', () => {
    const { getByText, toJSON } = render(<SbTitle>{testText}</SbTitle>);
    const textElement = getByText(testText);

    expect(textElement).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correct styles by default', () => {
    const { getByText } = render(<SbTitle>{testText}</SbTitle>);
    const textElement = getByText(testText);

    expect(textElement).toHaveStyle({
      fontFamily: theme.fonts.monserratBold,
      fontSize: theme.fontSize.s,
      color: theme.colors.primary,
    });
  });
});
