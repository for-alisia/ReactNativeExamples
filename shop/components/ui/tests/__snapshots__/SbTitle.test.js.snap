// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SbTitle UI Component renders a text 1`] = `
<Text
  style={
    Object {
      "color": "#DC8080",
      "fontFamily": "montserrat-bold",
      "fontSize": 16,
    }
  }
>
  Hello
</Text>
`;
